tests/foreman/sys/test_hooks.py

```diff
@@ -0,0 +1,159 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Foreman Hooks
+
+:Requirement: Foreman Hooks
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: SYS
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+
+"""
+from fauxfactory import gen_ipaddr
+from nailgun import entities
+from requests.exceptions import HTTPError
+from robottelo import ssh
+from robottelo.decorators import run_in_one_thread, tier3
+from robottelo.test import TestCase
+from robottelo.datafactory import valid_hosts_list, valid_hostgroups_list
+
+HOOKS_DIR = '/usr/share/foreman/config/hooks'
+LOGS_DIR = '/usr/share/foreman/tmp/hooks.log'
+
+
+@run_in_one_thread
+class ForemanHooksTestCase(TestCase):
+    """Test class for testing foreman hooks"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create logger script to be executed via hooks"""
+        super(ForemanHooksTestCase, cls).setUpClass()
+        cls.org = 1  # using Default_Organization
+        cls.script_path = "{}/logger.sh".format(HOOKS_DIR)
+        ssh.command(
+            '''printf '#!/bin/sh\necho "$(date): Executed $1 hook'''
+            + ''' on object $2" > {0}' > {1}'''.format(
+                LOGS_DIR, cls.script_path)
+        )
+        ssh.command("chmod 774 {}".format(cls.script_path))
+        ssh.command("chown foreman:foreman {}".format(cls.script_path))
+        ssh.command("restorecon -RvF {}".format(HOOKS_DIR))
+
+    @classmethod
+    def tierDownClass(cls):
+        """clean up hook files"""
+        super(ForemanHooksTestCase, cls).tierDownClass()
+        ssh.command("rm -rf {}/*".format(HOOKS_DIR))
+
+    @tier3
+    def test_positive_host_hooks(self):
+        """Create hooks to be executed on host create, update and destroy
+
+        :id: 4fe35fda-1524-44f7-9221-96d1aeafc75c
+
+        :Steps:
+            1. Create hook directories with symlinks to logger script
+            for subset of supported events: create, update, destroy
+            2. Perform trigger actions (create, update and delete host)
+            3. Observe custom logs created by hook script
+
+        :expectedresults: hook scripts are executed at proper time
+        """
+        host_name = valid_hosts_list()[0]
+        expected_msg = "Executed {0} hook on object {1}"
+
+        # create hook destination directories with logger script
+        create_event = "create"
+        update_event = "update"
+        destroy_event = "destroy"
+        for event in [create_event, destroy_event, update_event]:
+            hook_dir = "{0}/host/managed/{1}".format(HOOKS_DIR, event)
+            ssh.command('''mkdir -p {}'''.format(hook_dir))
+            ssh.command('''ln -sf {0} {1}/'''.format(self.script_path, hook_dir))
+        result = ssh.command("systemctl restart httpd")
```

Comment:
As I see multiple system calls.  Can we make this test as a destructive test?
What do you think?

Reply:
I thought that running in sequential tier would be safe enough, I don't mind moving it to the destructive tier if there is a call for it